includes/functions.php

```diff
@@ -2371,4 +2056,645 @@ function tsml_date_localised($format, $timestamp = null)
+            continue;
+        } elseif (in_array($import_key, $time_fields_to_check)) {
+            $db_value = date("H:i", strtotime($db_meeting[$import_key]));
+            $import_value = date("H:i", strtotime($import_meeting[$import_key]));
+            if ($db_value !== $import_value) {
+                if ($tsml_debug) {
+                    tsml_show_changed_values($import_meeting['name'], $import_key, $db_meeting[$import_key], $import_meeting[$import_key]);
+                }
+                return false;
+            }
+        } elseif (in_array($import_key, $fields_to_check_without_special_characters)) {
+            $db_value = (isset($db_meeting[$import_key])) ? tsml_remove_special_char($db_meeting[$import_key]) : '';
+            $import_value = (isset($import_meeting[$import_key])) ? tsml_remove_special_char($import_value) : '';
+            if ($db_value !== $import_value) {
+                if ($tsml_debug) {
+                    tsml_show_changed_values($import_meeting['name'], $import_key, $db_meeting[$import_key], $import_meeting[$import_key]);
+                }
+                return false;
+            }
+        } elseif (!empty($db_meeting[$import_key]) && array_key_exists($import_key, $import_meeting) && !empty($import_meeting[$import_key])) { //generic code to handle fields the same
+            $db_value = (isset($db_meeting[$import_key])) ? $db_meeting[$import_key] : '';
+            $import_value = (isset($import_meeting[$import_key])) ? $import_value : '';
+            if ($db_value !== $import_value) {
+                if ($tsml_debug) {
+                    tsml_show_changed_values('* ' . $import_meeting['name'], $import_key, $db_meeting[$import_key], $import_meeting[$import_key]);
+                }
+                return false;
+            }
+        }
+    }
+    //following fields are handled uniquely
+    if (!empty($db_meeting['post_title']) && !empty($import_meeting['name'])) {
+        $post_title = trim(tsml_remove_special_char($db_meeting['post_title']));
+        $import_name = trim(tsml_remove_special_char($import_meeting['name']));
+        if ($post_title !== $import_name) {
+            if ($tsml_debug) {
+                tsml_show_changed_values($import_meeting['name'], 'name', $db_meeting['post_title'], $import_meeting['name']);
+            }
+            return false;
+        }
+    }
+    if (!empty($db_meeting['post_name']) && !empty($import_meeting['slug'])) {
+        $db_value = $db_meeting['post_name'];
+        $import_value = $import_meeting['slug'];
+
+        if ($db_value !== $import_value) {
+            $data_source_ids = tsml_get_data_source_ids($data_source_url);
+            sort($data_source_ids);
+            $db_meetings = array_filter(tsml_get_meetings(), function ($meeting) use ($data_source_ids) {
```

Comment:
this is not efficient because it runs in a repeat loop, if we need all the data source meetings we should get them once and pass them in to `tsml_do_differential_comparison`

Reply:
Even  when a  slug might get changed once in a million records?